homeassistant/components/balboa/__init__.py

```diff
@@ -0,0 +1,88 @@
+"""The Balboa Spa Client integration."""
+import asyncio
+import time
+
+from pybalboa import BalboaSpaWifi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+import homeassistant.util.dt as dt_util
+
+from .const import _LOGGER, CONF_SYNC_TIME, DEFAULT_SYNC_TIME, DOMAIN, PLATFORMS
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Balboa Spa from a config entry."""
+    host = entry.data[CONF_HOST]
+
+    _LOGGER.debug("Attempting to connect to %s", host)
+    spa = BalboaSpaWifi(host)
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = spa
+
+    connected = await spa.connect()
+    if not connected:
+        _LOGGER.error("Failed to connect to spa at %s", host)
+        raise ConfigEntryNotReady
+
+    # send config requests, and then listen until we are configured.
+    await spa.send_mod_ident_req()
+    await spa.send_panel_req(0, 1)
+
+    _LOGGER.debug("Starting listener and monitor tasks")
+    hass.loop.create_task(spa.listen())
```

Comment:
Should the listen call happen after the callback is set?

Reply:
no, because it will fire instantly when listen() is started, and we haven't been added to hass, so its a nonetype.

Reply:
actually this does work in testing.